x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/privilege/ManageOwnApiKeyClusterPrivilegeTests.java

```diff
@@ -220,14 +247,28 @@ public void testCheckQueryApiKeyRequest() {
         final ClusterPermission clusterPermission = ManageOwnApiKeyClusterPrivilege.INSTANCE.buildPermission(ClusterPermission.builder())
             .build();
 
-        final QueryApiKeyRequest queryApiKeyRequest = new QueryApiKeyRequest();
+        final QueryApiKeyRequest queryApiKeyRequest1 = new QueryApiKeyRequest(null, null, null, null, null, randomBoolean());
         if (randomBoolean()) {
-            queryApiKeyRequest.setFilterForCurrentUser();
+            queryApiKeyRequest1.setFilterForCurrentUser();
         }
         assertThat(
-            clusterPermission.check(QueryApiKeyAction.NAME, queryApiKeyRequest, AuthenticationTestHelper.builder().build()),
-            is(queryApiKeyRequest.isFilterForCurrentUser())
+            clusterPermission.check(
+                QueryApiKeyAction.NAME,
+                queryApiKeyRequest1,
+                randomValueOtherThanMany(Authentication::isApiKey, () -> AuthenticationTestHelper.builder().build())
+            ),
+            is(queryApiKeyRequest1.isFilterForCurrentUser())
         );
+
+        // An API key cannot view limited-by role descriptors
+        final boolean withLimitedBy = randomBoolean();
+        final QueryApiKeyRequest queryApiKeyRequest2 = new QueryApiKeyRequest(null, null, null, null, null, withLimitedBy);
+        queryApiKeyRequest2.setFilterForCurrentUser();
+        assertThat(
+            clusterPermission.check(QueryApiKeyAction.NAME, queryApiKeyRequest2, AuthenticationTestHelper.builder().apiKey().build(false)),
+            is(false == queryApiKeyRequest2.isWithLimitedBy())
```

Comment:
I learnt to read `false == something` as `not something`. That is why I always have `false == ` in the beginning instead of having it at the end like `something == false`.

Reply:
Yup, I've started adopting this pattern as well.